plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/persistence/SerializableLibraryClasspathContainer.java

```diff
@@ -17,38 +17,56 @@
 package com.google.cloud.tools.eclipse.appengine.libraries.persistence;
 
 import com.google.cloud.tools.eclipse.appengine.libraries.LibraryClasspathContainer;
+import com.google.cloud.tools.eclipse.appengine.libraries.model.CloudLibraries;
+import com.google.cloud.tools.eclipse.appengine.libraries.model.Library;
+import com.google.cloud.tools.eclipse.appengine.libraries.model.LibraryFile;
+
 import java.util.ArrayList;
 import java.util.List;
 import org.eclipse.core.runtime.IPath;
 import org.eclipse.core.runtime.Path;
 import org.eclipse.jdt.core.IClasspathEntry;
+import org.eclipse.jdt.core.IJavaProject;
 
 /**
  * Represents a {@link LibraryClasspathContainer} in such a way that it can be easily transformed
  * into JSON.
  */
-public class SerializableLibraryClasspathContainer {
+class SerializableLibraryClasspathContainer {
 
   private final String description;
   private final String path;
   private final List<SerializableClasspathEntry> entries = new ArrayList<>();
+  private List<LibraryFile> libraryFiles;
 
-  public SerializableLibraryClasspathContainer(LibraryClasspathContainer container,
+  SerializableLibraryClasspathContainer(LibraryClasspathContainer container,
       IPath baseDirectory, IPath sourceBaseDirectory) {
     description = container.getDescription();
     path = container.getPath().toString();
 
     for (IClasspathEntry entry : container.getClasspathEntries()) {
       entries.add(new SerializableClasspathEntry(entry, baseDirectory, sourceBaseDirectory));
     }
+    
+    libraryFiles = new ArrayList<>(container.getLibraryFiles());
   }
 
-  public LibraryClasspathContainer toLibraryClasspathContainer(IPath baseDirectory,
+  LibraryClasspathContainer toLibraryClasspathContainer(IJavaProject javaProject, IPath baseDirectory,
       IPath sourceBaseDirectory) {
     List<IClasspathEntry> classpathEntries = new ArrayList<>();
+
     for (SerializableClasspathEntry entry : entries) {
       classpathEntries.add(entry.toClasspathEntry(baseDirectory, sourceBaseDirectory));
     }
-    return new LibraryClasspathContainer(new Path(path), description, classpathEntries);
+    
+    Library masterLibrary = CloudLibraries.getMasterLibrary(javaProject);
```

Comment:
Why is this general-purpose deserializer adding what could be arbitrary libraryFiles to the project's master library?

Reply:
It's actually not. I tried that but it hit some of the same issues as earlier. Instead it adds an empty list when it reads a legacy JSON (no libraryFiles element) so other parts of the code don't blow up when they get a null. This seems to be sufficient to avoid the problems on import you found yesterday.